app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -859,46 +1071,99 @@ private void openPopupPlayer(final boolean append) {
             return;
         }
 
-        final PlayQueue itemQueue = new SinglePlayQueue(currentInfo);
+        // See UI changes while remote playQueue changes
+        if (!bounded) startService(false);
+
+        //  If a user watched video inside fullscreen mode and than chose another player return to non-fullscreen mode
+        if (player != null && player.isInFullscreen()) player.toggleFullscreen();
+
+        PlayQueue queue = setupPlayQueueForIntent(append);
         if (append) {
-            NavigationHelper.enqueueOnPopupPlayer(activity, itemQueue, false);
+            NavigationHelper.enqueueOnPopupPlayer(activity, queue, false);
         } else {
-            Toast.makeText(activity, R.string.popup_playing_toast, Toast.LENGTH_SHORT).show();
-            final Intent intent = NavigationHelper.getPlayerIntent(
-                    activity, PopupVideoPlayer.class, itemQueue, getSelectedVideoStream().resolution, true
-            );
-            activity.startService(intent);
+            Runnable onAllow = () -> NavigationHelper.playOnPopupPlayer(activity, queue, true);
+            if (shouldAskBeforeClearingQueue()) showClearingQueueConfirmation(onAllow);
+            else onAllow.run();
```

Comment:
I see this piece of code is used three times, and `shouldAskBeforeClearingQueue()` is used only in those three places, so I think `shouldAskBeforeClearingQueue()` should be replaced by a function that does the same things as these two lines toghether. Something like (also note that I inverted the `player == null` condition and added `@Nullable final`; the function name is probably not optimal):
```java
void replaceQueueIfUserConfirms(final Runnable onAllow) {
    @Nullable final PlayQueue activeQueue = player == null ? null : player.getPlayQueue();

    // Player will have STATE_IDLE when a user pressed back button
    if (isClearingQueueConfirmationRequired(activity)
            && playerIsNotStopped()
            && activeQueue != null
            && !activeQueue.equals(playQueue)
            && activeQueue.getStreams().size() > 1) {
        showClearingQueueConfirmation(onAllow);
    } else {
        onAllow.run();
    }
}
```
And then this (and also the occourences below) would just become:
```suggestion
            replaceQueueIfUserConfirms(() -> NavigationHelper.playOnPopupPlayer(activity, queue, true));
```